<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>JointControlWidget</class>
 <widget class="QWidget" name="JointControlWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>683</width>
    <height>852</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Satus</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="6" column="2">
       <widget class="QLabel" name="label_20">
        <property name="text">
         <string>Debug Value 3</string>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QLCDNumber" name="debug3Display">
        <property name="digitCount">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QLCDNumber" name="externalADCDisplay"/>
      </item>
      <item row="5" column="2">
       <widget class="QLabel" name="label_19">
        <property name="text">
         <string>Debug Value 2</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QLCDNumber" name="debug1Display">
        <property name="digitCount">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item row="8" column="3">
       <widget class="QLCDNumber" name="angularVelocityDisplay"/>
      </item>
      <item row="7" column="3">
       <widget class="QLCDNumber" name="angularPositionDisplay"/>
      </item>
      <item row="2" column="3">
       <widget class="QLCDNumber" name="currentDisplay"/>
      </item>
      <item row="1" column="3">
       <widget class="QLCDNumber" name="voltageDisplay"/>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>External ADC</string>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QLCDNumber" name="debug2Display">
        <property name="digitCount">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Voltage</string>
        </property>
       </widget>
      </item>
      <item row="9" column="2" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="enableIcon">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="../resources/resources.qrc">:/icons/circle-check.svg</pixmap>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="errorIcon">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="../resources/resources.qrc">:/icons/exclamation-circle.svg</pixmap>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="movingIcon">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="../resources/resources.qrc">:/icons/run.svg</pixmap>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="calibratingIcon">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="../resources/resources.qrc">:/icons/current-location-off.svg</pixmap>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="calibratedIcon">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="../resources/resources.qrc">:/icons/direction-arrows.svg</pixmap>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="directionIcon">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="../resources/resources.qrc">:/icons/rotate-clockwise.svg</pixmap>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="8" column="2">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Angular Velocity</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Debug Value 1</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Current</string>
        </property>
       </widget>
      </item>
      <item row="7" column="2">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Angular Position</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Control</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="2">
       <widget class="QPushButton" name="enableButton">
        <property name="text">
         <string>Enable</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLCDNumber" name="commandDisplay">
        <property name="digitCount">
         <number>6</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Command</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="disableButton">
        <property name="text">
         <string>Disable</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QSpinBox" name="commandValueInput">
        <property name="minimum">
         <number>-100000</number>
        </property>
        <property name="maximum">
         <number>100000</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="3">
       <widget class="QScrollBar" name="commandValueSlider">
        <property name="minimum">
         <number>-100</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="tracking">
         <bool>true</bool>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="invertedControls">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QScrollArea" name="scrollArea">
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>657</width>
           <height>352</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="1">
           <widget class="QGroupBox" name="groupBox_5">
            <property name="title">
             <string>Command</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="0" column="0">
              <widget class="QComboBoxTimed" name="commandModeSelector"/>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string>Joint</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="3" column="1">
              <widget class="QSpinBoxTimed" name="legNumberInput">
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>Joint Type</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>Leg Number</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>Gear Ratio</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBoxTimed" name="gearRatioInput">
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
             <item row="7" column="1">
              <widget class="QSpinBoxTimed" name="nodeIdInput"/>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Node Id</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QComboBoxTimed" name="jointTypeSelector"/>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QGroupBox" name="groupBox_7">
            <property name="title">
             <string>Control</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_8">
             <item row="3" column="0">
              <widget class="QLabel" name="label_16">
               <property name="text">
                <string>Intergral Gain</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_17">
               <property name="text">
                <string>Proportional Gain</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_15">
               <property name="text">
                <string>Derivative Gain</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QDoubleSpinBoxTimed" name="proportionalGainInput">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimum">
                <double>-10000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QDoubleSpinBoxTimed" name="derivativeGainInput">
               <property name="minimum">
                <double>-10000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QDoubleSpinBoxTimed" name="integralGainInput">
               <property name="minimum">
                <double>-10000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QGroupBox" name="groupBox_8">
            <property name="title">
             <string>Calibratrion</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_9">
             <item row="0" column="0">
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>Minimum Angle</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBoxTimed" name="minAngleInput">
               <property name="minimum">
                <number>-10000</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_14">
               <property name="text">
                <string>Maximum Angle</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBoxTimed" name="maxAngleInput">
               <property name="minimum">
                <number>-10000</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_18">
               <property name="text">
                <string>Angle Offset</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBoxTimed" name="angleOffsetInput">
               <property name="minimum">
                <number>-10000</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QGroupBox" name="groupBox_6">
            <property name="title">
             <string>Telemetry</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Telemetry Period</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBoxTimed" name="telemetryPeriodInput">
               <property name="maximum">
                <number>60000</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QGroupBox" name="groupBox_9">
            <property name="title">
             <string>Motor</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_10">
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="zeroForceButton">
               <property name="text">
                <string>Zero Force</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="2">
              <widget class="QPushButton" name="debugButton">
               <property name="text">
                <string>Debug</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBoxTimed" name="pwmFrequencyInput">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>500</number>
               </property>
               <property name="value">
                <number>20</number>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_21">
               <property name="text">
                <string>PWM Freq (kHz)</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="resolutionTestButton">
               <property name="text">
                <string>Resolution Test</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QSpinBoxTimed</class>
   <extends>QSpinBox</extends>
   <header>qspinboxtimed.h</header>
  </customwidget>
  <customwidget>
   <class>QComboBoxTimed</class>
   <extends>QComboBox</extends>
   <header>qcomboboxtimed.h</header>
  </customwidget>
  <customwidget>
   <class>QDoubleSpinBoxTimed</class>
   <extends>QDoubleSpinBox</extends>
   <header>qdoublespinboxtimed.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
